Tests/SmithyRetriesTests/RetryIntegrationTests.swift

```diff
@@ -221,9 +227,10 @@ private class TestOutputHandler: ExecuteRequest {
         // Return either a successful response or a HTTP error, depending on the directions in the test step.
         switch testStep.response {
         case .success:
-            return HTTPResponse()
+            return HTTPResponse(statusCode: .ok)
         case .httpError(let statusCode):
-            throw TestHTTPError(statusCode: statusCode)
+            let httpStatusCode = HTTPStatusCode(rawValue: statusCode)!
+            return HTTPResponse(statusCode: httpStatusCode)
```

Comment:
Rather than throw here, successfully return an HTTP response that indicates a non-success error code.
Instead, the service error will get thrown in the deserializer above (this is how "real" clients work.)